#! /bin/sh /usr/share/dpatch/dpatch-run
## 17-no-newline-in-addmessage.dpatch by Michael Karcher
## <debian@mkarcher.dialup.fu-berlin.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Remove unsupported newlines from game messages. Important unless
## DP: signed character fix from 18-fix-drawichar is applied.

@DPATCH@
diff -urNad rott-1.0+dfsg~/rt_game.c rott-1.0+dfsg/rt_game.c
--- rott-1.0+dfsg~/rt_game.c	2008-01-10 09:19:03.000000000 +0100
+++ rott-1.0+dfsg/rt_game.c	2008-01-10 09:19:04.000000000 +0100
@@ -1777,7 +1777,7 @@
       GiveLives(1);
       if (ob==player)
          {
-         AddMessage("100 Life Item Points!  Extra Life!\n",MSG_BONUS);
+         AddMessage("100 Life Item Points!  Extra Life!",MSG_BONUS);
          SD_PlaySoundRTP (SD_GET1UPSND, player->x, player->y);
          }
       pstate->triads -= 100;
@@ -4184,7 +4184,7 @@
       killangle=startangle+(TOTALDEATHROT>>1);
       if (dopefish==true)
          {
-         AddMessage("Dopefish Death Cam\n",MSG_SYSTEM);
+         AddMessage("Dopefish Death Cam",MSG_SYSTEM);
          }
       for (iangle=startangle;;)
          {
